source/common/upstream/eds.cc

```diff
@@ -133,14 +138,17 @@ bool EdsClusterImpl::updateHostsPerLocality(const uint32_t priority, const HostV
   // out of the locality scheduler, we discover their new weights. We don't currently have a shared
   // object for locality weights that we can update here, we should add something like this to
   // improve performance and scalability of locality weight updates.
-  if (updateDynamicHostList(new_hosts, *current_hosts_copy, hosts_added, hosts_removed) ||
+
+  if (updateDynamicHostList(new_hosts, *current_hosts_copy, hosts_added, hosts_removed,
+                            new_all_hosts, all_hosts_) ||
```

Comment:
hmm why pass `all_hosts_` as a param given it's a member of `BaseDynamicClusterImpl` (so `updateDynamicHostList()` has access to it already)?

Reply:
good point! i'll update